File: Data Structures & Algorithms/subsets-ii/submission-1.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        
        res = []
        nums.sort()
        def helper(idx,subs):
                if idx == len(nums):
                    res.append(subs.copy())
                    return

                subs.append(nums[idx])
                helper(idx+1,subs)
                subs.pop()

                while idx+1<len(nums) and nums[idx] == nums[idx+1]:
                    idx+=1
                helper(idx+1,subs)
                return
        
        helper(0,[])
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        for (int i=0;i<nums.size();i++){
            if (i+1 < nums.size() and nums[i+1]==nums[i]){
                return true;
            }

        }
        return false;
    }
};